src/core/internal/hash.d

```diff
@@ -147,7 +147,9 @@ private template useScopeConstPassByValue(T)
     else static if (is(T == struct) || is(T == union))
     {
         // Overly restrictive for simplicity.
-        enum useScopeConstPassByValue = false;
+        enum useScopeConstPassByValue = T.sizeof <= (int[]).sizeof &&
```

Comment:
Why do you do this check?

Reply:
If `T.sizeof` is too large pass by value could lead to excessive data copying.